native.go

```diff
@@ -195,6 +195,20 @@ func (n NativeClass) InstanceOf(val interface{}) Value {
 	return obj
 }
 
+// CreateLazyObject creates a lazy object that will do initialization when the object is called upon during runtime
+func (r *Runtime) CreateLazyObject(create func(val *Object) *Object) (Value, error) {
+	if create == nil {
+		return UndefinedValue(), errors.New("create cannot be nil")
+	}
+
+	createObjectImpl := func(val *Object) objectImpl {
+		obj := create(val)
+		return obj.self
+	}
+
+	return r.newLazyObject(createObjectImpl), nil
+}
+
```

Comment:
While I know this is a trivial function we have gotten into the bad habit of not adding unit tests in goja. I'd like to change that, do you mind adding a simple table test for this function?

Reply:
hmm yeah, that's a good point! I can definitely do this 👍